scripts/check-release.js

```diff
@@ -0,0 +1,90 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const { version: releaseVersion } = require('../lerna.json')
+const { findPackages, printLog, logAndExit, executeCommand } = require('./utils')
+
+async function main() {
+  await checkGitTag()
+  await checkPackages()
+
+  printLog('Release check done.')
+}
+
+async function checkGitTag() {
+  printLog('Checking git tag')
+  const headRef = await executeCommand('git rev-parse HEAD')
+  let tagRef
+  try {
+    tagRef = await executeCommand(`git rev-list -n 1 v${releaseVersion} --`)
+  } catch (error) {
+    throw new Error(`Failed to find git tag reference: ${error}`)
+  }
+  if (tagRef !== headRef) {
+    throw new Error('Git tag not on HEAD')
+  }
+}
+
+async function checkPackages() {
+  const packages = await findPackages()
+
+  printLog(
+    `Checking packages:
+   - ${packages.map((pkg) => pkg.relativePath).join('\n   - ')}
+`
+  )
+
+  for (const pkg of packages) {
+    checkPackage(pkg)
+  }
+}
+
+function checkPackage(pkg) {
+  if (isBrowserSdkPackageName(pkg.manifest.name) && pkg.manifest.version !== releaseVersion) {
+    throw new Error(`Invalid version for ${pkg.relativePath}: expected ${releaseVersion}, got ${pkg.manifest.version}`)
+  }
+  checkPackageDependencyVersions(pkg)
+  checkPackageEntryPoints(pkg)
+}
+
+function checkPackageDependencyVersions(pkg) {
+  for (const dependencies of [pkg.manifest.dependencies, pkg.manifest.devDependencies, pkg.manifest.peerDependencies]) {
+    if (!dependencies) {
+      continue
+    }
+
+    for (const [dependencyName, dependencyVersion] of Object.entries(dependencies)) {
+      if (
+        isBrowserSdkPackageName(dependencyName) &&
+        !dependencyVersion.startsWith('file:') &&
+        dependencyVersion !== releaseVersion
+      ) {
+        throw new Error(
+          `Invalid dependency version for ${dependencyName} in ${pkg.relativePath}: expected ${releaseVersion}, got ${dependencyVersion}`
+        )
+      }
+    }
+  }
+}
+
+function checkPackageEntryPoints(pkg) {
+  for (const entryPointPath of [pkg.manifest.main, pkg.manifest.module, pkg.manifest.types]) {
+    if (!entryPointPath) {
+      continue
+    }
+
+    const absoluteEntryPointPath = path.resolve(pkg.path, entryPointPath)
+    if (!fs.existsSync(absoluteEntryPointPath)) {
+      throw new Error(
+        `Invalid entry point ${entryPointPath} in ${pkg.relativePath}: ${absoluteEntryPointPath} not found`
+      )
+    }
+  }
+}
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏are you checking entry points of all packages and not only `@datadog` packages in order to be sure to check `internal` packages?
if so, IMO it is a bit weird to check rum-events-format entry point as well.
Should we not consider browser sdk packages as packages under `packages` directory and adjust other checks accordingly?

Reply:
I found out that in `check-licenses` we are iterating over `rum-events-format` as well, so I kept this behavior, and since it does not matter that much to check for those packages as well in `check-release`, I didn't filter them out.

But sure, I can remove  packages from `rum-events-format` if you think that it's not needed. I think we should do the same for `check-licenses` then. WDYT?

>  adjust other checks accordingly?

I'm not sure what you mean by that

Reply:
> But sure, I can remove packages from rum-events-format if you think that it's not needed. I think we should do the same for check-licenses then. WDYT?

sounds good to me

>> Should we not consider browser sdk packages as packages under packages directory and adjust other checks accordingly?

> I'm not sure what you mean by that

it seemed to me that some of the checks performed in `checkPackage` would need need to be updated if we change what we consider to be browser sdk packages.